perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -780,7 +780,7 @@ public RichMember getRichMemberWithAttributes(PerunSession sess, Member member)
 		return richMembersWithAttributes;
 	}
 
-	public List<RichMember> getRichMembersWithAttributesByNames(PerunSession sess, Group group, Resource resource, List<String> attrsNames) throws InternalErrorException, WrongAttributeAssignmentException, AttributeNotExistsException {
+	public List<RichMember> getRichMembersWithAttributesByNames(PerunSession sess, Group group, Resource resource, List<String> attrsNames) throws InternalErrorException, AttributeNotExistsException, MemberResourceMismatchException {
```

Comment:
I guess checking, that group and resource are from the same VO, would be more interesting. Since when it's true, there is no chance, that member wouldn't be from the same VO as resource. Hence you should catch ``MemberResourceMismatchException`` thrown by conversion method and wrap it as ``ConsistencyErrorException``. Then you can add explicit check on group and resource and throw necessary exception to API layer.